src/core/server/http/router/socket.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Socket } from 'net';
+import { DetailedPeerCertificate, PeerCertificate, TLSSocket } from 'tls';
```

Comment:
__question:__ not related to PR, do we have any central place in Kibana where we can patch NodeJS types (`PeerCertificate` type doesn't expose `fingerprint256` right now even though such property exists and used by PKI provider)? If not, I'll probably send PR to DefinitelyTyped later.

Reply:
No. I'd expect such declaration under appropriate domain folder (next to KibanaSocket) as a temporary solution. In the long-term, it's better to update DefinitelyTyped.